### PR TITLE
Implement container reordering

### PR DESCRIPTION
**Pull request by sryze**
This is just a small patch that makes it possible to move / reorder containers via drag&drop. 

I added a third button next to the edit (pencil) button that allows users to drag a container and insert it before another container. Container positions are saved in the browser's local storage.

I got the icon from the [Mozilla Foundation Icon Font](https://mozilla.github.io/foundation-icons/).

Related issue: #330 

Example:

firefox-container-reordering

[firefox_multi-account_containers-6.1.1.zip](https://github.com/mozilla/multi-account-containers/files/4007793/firefox_multi-account_containers-6.1.1.zip)

### CODE REVIEW

File: src/img/container-move.svg
Comment (jonathanKingston) [inline src/img/container-move.svg]: Please remove this comment. Where was this image from also? Is there any copyrights on it?

File: src/js/popup.js
Comment (jonathanKingston) [inline src/js/popup.js]: This doesn't persist the sorted state anywhere does it?

File: src/js/popup.js
Comment (jonathanKingston) [inline src/js/popup.js]: Can you make this part of the `promise.all` above and then merge it into `_identities` in the `map()`?

Comment (jonathanKingston) [reply src/js/popup.js]: I suspect we should treat this code as if there is always an order rather than make it a little more complex if there is an optional sort order. I also think to make it deterministic we should always add a sort order when a new container is added.

File: src/js/popup.js
Comment (jonathanKingston) [inline src/js/popup.js]:
```suggestion
      tr.addEventListener("dragstart", (e) => {
```
Just to remain consistent with other code in this repo

File: src/js/popup.js
Comment (jonathanKingston) [inline src/js/popup.js]: This part from 961-969 of saving the order feels like it should be a separate function probably.

File: src/js/popup.js
Comment (jonathanKingston) [inline src/js/popup.js]: No inline styles please

File: src/js/popup.js
Comment (jonathanKingston) [inline src/js/popup.js]: I'd prefer we stored the sort order in the state key rather than have another duplicated array with cookiestoreid in it.

Comment (sryze) [reply src/js/popup.js]: What do you mean by "state key"? are you talking about the state received from `queryIdentitiesState`?

**Comment by sryze** [comment]
@jonathanKingston I fixed some stuff according to your suggestions, please take a look

> Would you be interested in implementing this also into firefox as a follow up? (I'm happy to help mentor this).
I'm not sure, how does that work? do I create a PR in the Firefox repo and add you as a reviewer?

**Comment by Laurensdc** [comment]
This needs to happen

**Comment by idoleat** [comment]
I would be much appreciated if this PR is approved :pray: Thanks for your awesome job!

File: src/js/popup.js
Comment (sryze) [reply src/js/popup.js]: Fixed this

File: src/js/popup.js
Comment (sryze) [reply src/js/popup.js]: Fixed

File: src/js/popup.js
Comment (sryze) [reply src/js/popup.js]: done

File: src/img/container-move.svg
Comment (sryze) [reply src/img/container-move.svg]: Removed

**Comment by jwdevel** [comment]
Query: will the new order chosen here be reflected when the "sort tabs" button is clicked?

At least from my previous experimenting, changing the order in containers.json does *not* affect that other ordering, and I wonder if this is any different.

**Comment by builtbyjordan** [comment]
Bump on this! It would certainly be very helpful. I'll even help out with testing if need be.  Thanks for this.  Hopefully it gets approved soon.

**Comment by aghassemlouei** [comment]
Any progress on approvals?

**Comment by lindhe** [comment]
@jonathanKingston What's the review process like? Can you approve this, or do we need someone else from Mozilla to look at it too?

**Comment by MurzNN** [comment]
Maybe @maxxcrawford can help with reviewing and merging, or comment situation? He do last merge recently.

**Comment by nicolasbadia** [comment]
Anyone to approve this? Can we help?

**Comment by heitorPB** [comment]
There are conflicts in `src/css/popup.css` and in `src/js/popup.js`.

@sryze, could you rebase to fix this?

**Comment by gethubb** [comment]
This looks great! Hope you're able to do the fix @sryze and get the approval... good work!

**Comment by maxxcrawford** [comment]
@MurzNN Thanks for the tag. Reviewed!

**Comment by sryze** [comment]
@madalincm I will try to fix the conflicts today.

**Comment by urda** [comment]
> @madalincm I will try to fix the conflicts today. 

Your efforts are appreciated￼

**Comment by sryze** [comment]
Merged with master, in the new version buttons will be shown inside the "Manage Containers" menu:

firefox-container-reordering

Also I couldn't get the top black border effect to work for some reason, so now containers are just highlighted in gray when drag over them.

**Comment by lindhe** [comment]
@maxxcrawford It says that you have requested changed, but I see no lingering comments. Perhaps you need to review again?

**Comment by Makishima** [comment]
Can we have this feature, please?

**Comment by mahalel** [comment]
Please can we have this feature, I have over 30 containers now and it's getting hard to find the right one when needed :(

**Comment by urda** [comment]
I'm willing to donate 20 bucks to Mozilla directly when this feature is merged.

**Comment by urda** [comment]
What can I do to help support this!

**Comment by BugReportering** [comment]
Why isn't this implemented yet?  We all need this, OBVIOUSLY!!  :100: :-D
Thank you @sryze for your time spent on this project!

**Comment by lindhe** [comment]
Since this is getting close to a year since publication, I would very much appreciate a statement from Mozilla explaining why this is not getting any traction. Is the feature unwanted or is it hard to merge for some technical reason? In the world of CI/CD, this seems to lack the C.

**Comment by urda** [comment]
What's the status of this PR? Can we get this feature in 2021?

**Review by maxxcrawford — CHANGES_REQUESTED** [review]
@sryze This is ready to merge in, pending the dark mode we added today. Could you please resolve the conflicts one more time? After that, we can add this in!

**Comment by duoi** [comment]
@maxxcrawford it's 2021, you get paid to do this, after dragging your knuckles for a literal year why don't you just fix the conflicts and merge it in?

**Comment by jc00ke** [comment]
@duoi hey dude, that's super rude and uncalled for. A lot of us share in your frustration but that's no way to talk to someone in OSS.

**Comment by groovecoder** [comment]
Thank you @jc00ke.

@duoi - this project's [Code of Conduct](https://github.com/mozilla/multi-account-containers/blob/master/CODE_OF_CONDUCT.md) is the [Mozilla Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/) which state:

> frustration should never turn into a personal attack

Don't make statements that personally attack a member of this project, or you will be banned from participation.

**Comment by sryze** [comment]
> @sryze This is ready to merge in, pending the dark mode we added today. Could you please resolve the conflicts one more time? After that, we can add this in! _(But also, be sure this looks/works as expected in dark mode)_

Done

**Comment by duoi** [comment]
@groovecoder didn't personally attack anybody, I literally just asked why he doesn’t fix the merge conflict himself. That’s not a personal attack any more than this comment is lol. Anyway, let’s not derail.

**Review by groovecoder — APPROVED** [review]
Code and spot-check looks good.

**Comment by groovecoder** [comment]
(I tested in dark mode.)

**Review by maxxcrawford — APPROVED** [review]
@sryze This is great. Merging in now. Thanks for all the hard work on this.

**Comment by urda** [comment]
When will this release to production?

**Comment by nicolasbadia** [comment]
@urda It already is released to production.

**Comment by urda** [comment]
@nicolasbadia  Ah guess my add-ons haven't updated yet! Just got the latest and greatest and re-ordering my containers!

As promised, donation sent:

image

**Comment by jc00ke** [comment]
I'm so happy! Thank you all so much for making this happen. Was this issue open a long time? Sure. Do I give a sh.... not anymore! I really appreciate the push towards the end from those responsible for code review and getting this into production. :pray: :pray: :pray:

**Comment by maxxcrawford** [comment]
@urda If you add-ons are not set to update automatically, you can update them by going to `about:addons` and clicking on the gear icon and selecting "Check for updates" 

image

**Comment by placoderm** [comment]
I am running 8.0.6 version of the plugin and I don't see the handles to re-arrange the containers.
Any idea what I am doing wrong? FF 98.0.1
image

Sorry if this isn't the right place to ask this question.

**Comment by Vinnl** [comment]
Hi @placoderm, your screenshot is from Firefox's settings menu, whereas this updated affected the extension menu (install the extension [from here](https://addons.mozilla.org/firefox/addon/multi-account-containers/) if you don't have it yet). You can then re-order containers by opening the menu ([pictured here](https://addons.mozilla.org/user-media/previews/full/266/266158.png?modified=1643906676)) and clicking "Manage Containers" at the bottom.

**Comment by placoderm** [comment]
@Vinnl Thanks for the super fast reply.

I wasn't seeing that at the bottom of the list. I had to zoom waaaaay out to see it. 

I have moved one of the containers to the top (the yellow sun glasses)
image

However, when I press and hold the `+` button to open a new container, the possition hasn't changed. You can see it still in its original place:

image
I have closed and restarted FF. 

Any ideas?

**Pull request merged**
